code/modules/assembly/infrared.dm

```diff
@@ -246,7 +246,9 @@
 	hit()
 
 /obj/effect/beam/i_beam/Crossed(atom/movable/AM as mob|obj)
-	if(istype(AM, /obj/effect/beam) || !AM.density)
+	if(!isobj(AM) && !isliving(AM))
+		return
+	if(istype(AM, /obj/effect/beam))

```

Comment:
Both this and the proximity check should probably just exclude `/obj/effect`; I doubt there's any effects that should actually trip them, but there's plenty that shouldn't.